### PR TITLE
Changes for revng-distributable-revamp

### PR DESCRIPTION
**Pull request by mrjackv**

### CODE REVIEW

File: python/revng/cli/_commands/daemon_self_test.py [hunk 1]
```diff
@@ -83,9 +83,8 @@ def log(string: str):
     def fail(self, ex: Exception):
         if self.process is not None:
             self.process.terminate()
-            self.process.communicate()
-            if self.process.stdout is not None:
-                self.log(f"\n\n Daemon Stacktrace:\n{self.process.stdout.read()}")
+            stdout = self.process.communicate()[0]
+            self.log(f"\n\nDaemon Stacktrace:\n{stdout}")
```
Comment (aleclearmind) [inline python/revng/cli/_commands/daemon_self_test.py:87]: `"\n\nrevng-daemon output:\n`

File: lib/Support/PathList.cpp [hunk 2]
```diff
@@ -35,7 +35,11 @@ std::string getCurrentRoot() {
   Maps.split(MapsLines, "\n");
 
   for (const auto &Line : MapsLines) {
-    llvm::StringRef File = std::get<1>(Line.rsplit(" "));
+    llvm::SmallVector<llvm::StringRef> Parts;
+    Line.split(Parts, ' ', 5);
+    revng_assert(Parts.size() == 6, "Malformed /proc/self/maps");
+
+    llvm::StringRef File = Parts[5].ltrim();
```
Comment (aleclearmind) [inline lib/Support/PathList.cpp:42]: Mmmh, this does not fix the problem described in the commit message.
You should get anything that comes after the 4th space.

**Review by aleclearmind — CHANGES_REQUESTED** [review]

File: lib/Support/PathList.cpp (hunk 2 above)
Comment (aleclearmind) [reply lib/Support/PathList.cpp:42]: Ah-ah! Didn't know that! My bad!

File: python/revng/cli/_commands/daemon_self_test.py (hunk 1 above)
Comment (mrjackv) [reply python/revng/cli/_commands/daemon_self_test.py]: Fixed

**Comment by aleclearmind** [comment]
Merged in 91c4bdc. Thanks!

**Pull request closed without merging**